opds.py

```diff
@@ -1432,15 +1432,19 @@ def add_breadcrumbs(self, lane, include_lane=False, entrypoint=None):
         site_root_lane = None
         usable_parentage = []
         if lane is not None:
-            for ancestor in lane.parentage:
+            for ancestor in [lane] + list(lane.parentage):
```

Comment:
While testing against Open eBooks I discovered that my original implementation doesn't work if you ask for breadcrumbs for a lane that is, itself the root lane for some patron type. `lane.parentage` doesn't include the lane itself, so we never ran the `ancestor.root_for_patron_type` against the lane we were looking at.